api/api.proto

```diff
@@ -1203,6 +1203,83 @@ service OpenStorageRole {
       };
     }
 }
+
+// OpenStorageFilesystemTrimService provides methods to manage a
+// filesystem trim operation on a volume. This operation runs in the
+// background on a mounted volume. If the volumes are not mounted, these API 
+// return error.
+//
+service OpenStorageFilesystemTrimService {
+
+  // Start a filesystem Trim background operation on a mounted volume
+  rpc Start(SdkFilesystemTrimStartRequest)
+    returns (SdkFilesystemTrimStartResponse){
+      option(google.api.http) = {
+        post: "/v1/filesystem-trim/start"
+        body: "*"
+      };
+    }
+
+  // Get Status of a filesystem Trim background operation on a mounted volume, if any
+  rpc GetStatus(SdkFilesystemTrimGetStatusRequest)
+    returns (SdkFilesystemTrimGetStatusResponse){
+      option(google.api.http) = {
+        get: "/v1/filesystem-trim/getstatus/query/{volume_id}/{mount_path}"
+      };
+    }
+
+  // Stop a filesystem Trim background operation on a mounted volume, if any
+  rpc Stop(SdkFilesystemTrimStopRequest)
+    returns (SdkFilesystemTrimStopResponse){
+      option(google.api.http) = {
+        post: "/v1/filesystem-trim/stop"
+        body: "*"
+      };
+    }
+  }
+
+// OpenStorageFilesystemCheckService service provides methods to manage a 
+// filesystem check operation on a volume. This operation is run in the 
+// background on an unmounted volume. If the volumes are mounted, then these
+// APIs return error.
+service OpenStorageFilesystemCheckService {
+
+  // Get a report of issues found on the filesystem. This operation works on an unmounted volume.
+  rpc Report(SdkFilesystemCheckReportRequest)
+    returns (SdkFilesystemCheckReportResponse){
+      option(google.api.http) = {
+        post: "/v1/filesystem-check/report"
+        body: "*"
+      };
+    }
+
+  // Fix all issues reported in the response to ReportIssues API on a filesystem. This operation works on an unmounted volume.
+  rpc Fix(SdkFilesystemCheckFixRequest)
+    returns (SdkFilesystemCheckFixResponse){
+      option(google.api.http) = {
+        post: "/v1/filesystem-check/fix"
+        body: "*"
+      };
+    }
+
+  // Get Status of a filesystem check background operation on an unmounted volume, if any
+  rpc GetStatus(SdkFilesystemCheckGetStatusRequest)
+    returns (SdkFilesystemCheckGetStatusResponse){
+      option(google.api.http) = {
+        get: "/v1/filesystem-check/getstatus/query/{volume_id}"
```

Comment:
This can be `/v1/filesystem-check/getstatus/{volume_id}`

Reply:
changed it a to post request

Reply:
As I've pointed out in previous PRs (where it is also done incorrectly), we shouldn't be using PUT or POST for read APIs.
This is clearly explained here: https://www.w3schools.com/tags/ref_httpmethods.asp

Also, you shouldn't have the verb in the path itself. It should be a `GET`  on `/v1/filesystem-check/{volume_id}/status`

Reply:
I have removed "get" verb from the REST API path

There are 2 parameters expected for this API
1. volumeid
2. mountpath
Having mountpath as a query parameter for get would require the client to encode the request
we decided it would be easier for the client and also from the server to deal with mountpath as parameter given as part of the POST request

Let me know if you still insist that we need this API to use the GET verb

If you s

Reply:
discussed with dinesh. Finally we concluded to use GET request instead of POST for all the status APIs